internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -1216,10 +1219,103 @@ func (c *Coordinator) generateComponentModel() (err error) {
 	// If we made it this far, update our internal derived values and
 	// return with no error
 	c.derivedConfig = cfg
+	c.lastComponentModel = c.componentModel
 	c.componentModel = comps
+
+	// don't run a bunch of relatively costly code unless we're in debug mode
+	if c.logger.IsDebug() {
+		c.checkAndLogUpdate()
+	}
```

Comment:
IMO this is valuable even when not in debug mode, policy changes aren't that frequent so the actual amortized cost of this should be low.

I'd vote we turn this on by default.

Reply:
Yah, seems useful, but I was kinda worried about overhead, particularly if we start adding info on component updates. Should probably run some benchmarks and see what it looks like.